WcaOnRails/app/models/competition.rb

```diff
@@ -41,22 +41,27 @@ class Competition < ActiveRecord::Base
 
   # We have stricter validations for confirming a competition
   [:cityName, :countryId, :venue, :venueAddress, :website, :latitude, :longitude].each do |field|
-    validates field, presence: true, if: :isConfirmed?
+    validates field, presence: true, if: :confirmed_or_visible?
   end
```

Comment:
Maybe some of things I'm commenting aren't related to your PR, but since they show up here I feel like changing them =) So here I would see
```ruby
validates :cityName, :countryId, :venue, :venueAddress, :website, :latitude, :longitude, presence: true, if: :confirmed_or_visible?
```